worker-markup-core/src/main/java/com/github/cafdataprocessing/worker/markup/core/XmlConverter.java

```diff
@@ -171,16 +166,23 @@ private static void appendEmailHeaderToBuilder(StringBuilder headersBuilder, Str
         if(headerValues.isEmpty()){
             LOG.info("No '" + headerSourceFieldName + "' values to add in email headers during markup.");
         }
-        else if (headerValues.size() > 1) {
-            throw new AddHeadersException("Unable to add email headers. Multiple field entries found for header: "
-                    +headerSourceFieldName);
-        }
         else{
+            boolean isPriorityHeader = headerSourceFieldName.equals("priority");
+
             headersBuilder.append(headerStartText+": ");
-            String headerValue = headerSourceFieldName.equals("priority")
-                    ? Normalizations.normalizePriority(headerValues.get(0))
-                    : headerValues.get(0);
+            Iterator headersIterator = headerValues.iterator();
+            String headerValue = isPriorityHeader == true
+                    ? Normalizations.normalizePriority((String)headersIterator.next())
+                    : (String)headersIterator.next();
             headersBuilder.append(headerValue);
+            while(headersIterator.hasNext())
```

Comment:
- Java 8 has `String.join()`, which you could use instead of this loop.
- It also has `Collectors.joining()` which you could use if you wanted to combine with the priority normalisation like you are currently doing - i.e. something like:

        Stream<String> headerValuesStream = headerValues.stream();
        if (isPriorityHeader) {
            headerValuesStream = headerValuesStream.map(Normalizations::normalizePriority);
        }
        final String headerValue = headerValuesStream.collect(Collectors.joining("; "));

Reply:
Updated to use String.join().